### ISSUE TITLE
web_hook custom headers not getting processed

### ISSUE THREAD

**Issue by Musarrath** [jhuckaby/Cronicle#486]
## Summary

Trying for sending custom headers on webhook callback on event trigger. But webhook event is not getting triggered with the custom headers as per this [link](https://github.com/jhuckaby/Cronicle#event-web-hook ).

Sample [webhook site url](https://webhook.site/d378d801-e6b8-4bf6-9111-c67dfc0ecd7b) used for testing.

## Steps to reproduce the problem
Event request body

```json
{
  "plugin": "urlplug",
  "title": "Webhook Test",
  "enabled": 1,
  "algo": "random",
  "category": "general",
  "target": "allgrp",
  "timing": {
    "years": [
      2022
    ],
    "months": [
      4
    ],
    "days": [
      28
    ],
    "hours": [
      18
    ],
    "minutes": [
      30
    ]
  },
  "timezone": "Etc/UTC",
  "params": {
    "method": "POST",
    "timeout": "30",
    "headers": "Content-Type: application/json\nkey: c99c3a85",
    "data": "{\"message\":\"test message\"}",
    "url": "http://service.endpoint/call"
  },
  "web_hook": "https://webhook.site/d378d801-e6b8-4bf6-9111-c67dfc0ecd7b [header: key: test]"
}
```

In web_hook, custom headers are not getting processed as expected. The custom header not sent to webhook endpoint on event trigger.

## Your Setup
Used below command for local setup

```
curl -s https://raw.githubusercontent.com/jhuckaby/Cronicle/master/bin/install.js | node
```

### Operating system and version?
MacOS Monterey 12.1

### Node.js version?
v12.22.8

### Cronicle software version?
0.9.2

### Are you using a multi-server setup, or just a single server?
Single server setup 

### Are you using the filesystem as back-end storage, or S3/Couchbase?
filesystem as back-end storage - local
S3 - on server

### Can you reproduce the crash consistently?
yes

## Log Excerpts

```
[1648727504.733][2022-03-31 17:21:44][admins-macbook-pro.local][77475][Cronicle][debug][9][Web Hook Error: https://webhook.site/d378d801-e6b8-4bf6-9111-c67dfc0ecd7b [header: key: test]: Error: HTTP 404 Not Found][]
[1648727505.222][2022-03-31 17:21:45][admins-macbook-pro.local][77475][Cronicle][debug][9][Web Hook Error: https://webhook.site/d378d801-e6b8-4bf6-9111-c67dfc0ecd7b [header: key: test]: Error: HTTP 404 Not Found][]
```

**Comment by jhuckaby** [comment]
Node v12 is pretty old at this point.  We only officially support the LTS version of Node (currently v16) or newer.  Could you try upgrading and repeat your test?

Also, beyond Node.js itself, I suspect the problem may be a previous install of Cronicle and its supporting NPM libraries that weren't properly upgraded for some reason.  The underlying request library Cronicle uses, [pixl-request](https://github.com/jhuckaby/pixl-request), had the header injection feature added in [v1.0.33](https://github.com/jhuckaby/pixl-request/commit/4c3b1eedd478c90638827f0d03d949086af9460c), which was released on Nov 8, 2020.  I suspect you may have an older version.

Try these steps (after upgrading Node.js to v16):

```
sudo bash
cd /opt/cronicle
bin/control.sh stop
rm -rf node_modules
npm install
bin/control.sh start
```

**Comment by Musarrath** [comment]
Hi Jhuckaby,

Thanks for the reply, I tried changing the version and all, now its taking the headers but url is calling like this only.

```
https://webhook.site/d378d801-e6b8-4bf6-9111-c67dfc0ecd7b [header: key: test] (or)
[https://webhook.site/d378d801-e6b8-4bf6-9111-c67dfc0ecd7b]()%20[header:%20key:%20c99c3a85]
```

**Comment by jhuckaby** [comment]
Yup, you're absolutely right @Musarrath.  That's broken!

Fixed in [v0.9.3](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.3).  Thank you!

-- Joe

**Comment by Musarrath** [comment]
Hi @jhuckaby,

Thanks for the fix, It's working fine now.

I have one doubt wants to clarify it.
I am using single server as primary server but if the server crashed and If I wants the server to restart automatically is there any options to automatically restart it or shall I have to use the multiple server only if the primary server crashed then the other server will work as primary server as you mentioned in the documentation.
